### PR TITLE
Update docs unit testing typescript

### PR DESCRIPTION
**Pull request by soosgyul**

### CODE REVIEW

File: docs/docs/unit-testing.md [hunk 1]
```diff
@@ -229,16 +229,13 @@ Then update the configuration in `jest.config.js`, like so:
...
-    "^.+\\.jsx?$": "<rootDir>/jest-preprocess.js",
+    "^.+\\.[jt]sx?$": "<rootDir>/jest-preprocess.js",
   },
-  testRegex: "(/__tests__/.*|(\\.|/)(test|spec))\\.([tj]sx?)$",
   moduleNameMapper: {
     ".+\\.(css|styl|less|sass|scss)$": "identity-obj-proxy",
     ".+\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$":
       "<rootDir>/__mocks__/file-mock.js",
   },
-  moduleFileExtensions: ["ts", "tsx", "js", "jsx", "json", "node"],
```
Comment (pieh) [inline docs/docs/unit-testing.md]: I see that `jest` have our current/previous `testRegex`/`moduleFileExtensions` settings as default now:

https://jestjs.io/docs/en/configuration.html#modulefileextensions-array-string
> moduleFileExtensions [array]
> Default: ["js", "json", "jsx", "ts", "tsx", "node"]

https://jestjs.io/docs/en/configuration.html#testregex-string-array-string
> testRegex [string | array]
> Default: (/__tests__/.*|(\\.|/)(test|spec))\\.[jt]sx?$

Below section you added about updating `jest.preprocess.js` there are 2 paragraphs explaining those setting fields which now seem redundant, so maybe we should remove them?

Comment (soosgyul) [reply docs/docs/unit-testing.md]: Good spot. I've removed those two paragraphs and pushed the changes.

File: docs/docs/unit-testing.md [hunk 2]
```diff
@@ -249,9 +246,15 @@ module.exports = {
...
-contain tests. The pattern above matches any `.js`, `.jsx`, `.ts` or `.tsx`
+Also update `jest.preprocess.js` with the following babel preset to look like this:
+
+```js:title=jest-preprocess.js
+const babelOptions = {
+  presets: ["babel-preset-gatsby", "@babel/preset-typescript"],
+}
+```
+
+The pattern above matches any `.js`, `.jsx`, `.ts` or `.tsx`
```
Comment (pieh) [inline docs/docs/unit-testing.md]: We don't have "pattern above" anymore, so if we chose to keep information what tests will be picked up we should mention that `jest` by default will use those instead and link to `jest` config docs

Comment (soosgyul) [reply docs/docs/unit-testing.md]: I went ahead and did a bigger cleanup and removed more redundant references.

**Comment by pieh** [comment]
This looks so much cleaner! One thing that I didn't think about before is that this assumes that `@babel/preset-typescript` is installed (which is dependency of `gatsby-plugin-typescript`, so if user use that, they will have this package installed). Given that this babel preset is being used in configuration probably worth leaving a note about this somewhere near the beginning of typescript section.

File: docs/docs/unit-testing.md [hunk 3]
```diff
@@ -217,50 +217,24 @@ by running `npm test -- -u`.
...
+If you are using TypeScript, you need to make two small changes to your
+config.
 
-```shell
-npm install --save-dev ts-jest
+Update the transform in `jest.config.js`, like so:
+
+```js:title=jest.config.js
+    "^.+\\.[jt]sx?$": "<rootDir>/jest-preprocess.js",
 ```
```
Comment (marcysutton) [inline docs/docs/unit-testing.md]: I'd also add an example here to remove any ambiguity:
````suggestion

For example, if your test utilities are centrally located in a `lib/` directory in your Gatsby site:

```js:title=jest.config.js
    "^.+\\.[jt]sx?$": "lib/jest-preprocess.js",
```
````

File: docs/docs/unit-testing.md [hunk 4]
```diff
@@ -217,50 +217,24 @@ by running `npm test -- -u`.
...
 
-Then update the configuration in `jest.config.js`, like so:
+Also update `jest.preprocess.js` with the following babel preset to look like this:
```
Comment (marcysutton) [inline docs/docs/unit-testing.md]:
```suggestion
Also update `jest.preprocess.js` with the following Babel preset to look like this:
```

File: docs/docs/unit-testing.md (hunk 3 above)
Comment (soosgyul) [reply docs/docs/unit-testing.md]: The test files don't have to live inside the lib folder. In my project where I've applied the changes all my components are in a separate folder, and all tests live inside this folder...

Comment (ascorbic) [reply docs/docs/unit-testing.md]: I think there's a misunderstanding here. The user isn;t supposed to replace `<rootDir>` with their own path: the jest config expects the literal string `<rootDir>`. which is replaced *by Jest* with the root directory of the project. e.g. see https://github.com/gatsbyjs/gatsby/blob/master/jest.config.js#L37

Comment (marcysutton) [reply docs/docs/unit-testing.md]: Ok, that is definitely confusing as angle brackets often denote "replace this" in docs. You can disregard my example, but we should clarify that the literal string `<rootDir>` needs to be included.

File: docs/docs/unit-testing.md [hunk 5]
```diff
@@ -217,50 +217,24 @@ by running `npm test -- -u`.
...
 ## Using TypeScript
 
-If you are using TypeScript, you need to make a couple of small changes to your
-config. First install `ts-jest`:
+If you are using TypeScript, you need to make two changes to your
+config.
 
-```shell
-npm install --save-dev ts-jest
+Update the transform in `jest.config.js` to run `jest-preprocess` on files in your project's root directory, where `<rootDir>` is replaced with the correct relative path:
```
Comment (marcysutton) [inline docs/docs/unit-testing.md]:
```suggestion
Update the transform in `jest.config.js` to run `jest-preprocess` on files in your project's root directory, where `<rootDir>` is kept intact as a literal string to dynamically pass the relative path to Jest:
```
Is this accurate to communicate how the `<rootDir>` string is used here? (this was already updated once, based on my own confusion)

Comment (soosgyul) [reply docs/docs/unit-testing.md]: Yes, I think this is correct. Sorry I meant to update this all week, never got to it. :/

**Comment by soosgyul** [comment]
I'm doubtful that my update to the documentation made the windows unit tests fail.

File: docs/docs/unit-testing.md [hunk 6]
```diff
@@ -217,50 +217,24 @@ by running `npm test -- -u`.
...
 ## Using TypeScript
 
-If you are using TypeScript, you need to make a couple of small changes to your
-config. First install `ts-jest`:
+If you are using TypeScript, you need to make two changes to your
+config.
 
-```shell
-npm install --save-dev ts-jest
+Update the transform in `jest.config.js` to run `jest-preprocess` on files in your project's root directory, where `<rootDir>` is kept intact as a literal string to dynamically pass the relative path to Jest:
```
Comment (muescha) [inline docs/docs/unit-testing.md]: i suggest more easy english:

```suggestion
Update the transform in `jest.config.js` to run `jest-preprocess` on files in your project's root directory. 

**Note:** `<rootDir>` is replaced by Jest with the root directory of the project. Don't change it.
```

**Comment by laurieontech** [comment]
This looks great! Thanks for the changes.

**Pull request merged**